web-module/src/main/webapp/resources/structEditJS/dateTime/celements-date-time-field.js

```diff
@@ -24,168 +24,164 @@
-
-      _onChanged: function() {
-        const _me = this;
-        const newValue = _me.getValue();
-        console.debug("_onChanged", newValue);
-        const validatedValue = _me._fieldValidator(newValue);
-        _me._inputField.classList.toggle('validation-failed', !validatedValue);
-        if (newValue !== validatedValue) {
-          _me._inputField.value = validatedValue;
-        } else {
-          _me.celFire(_me.FIELD_CHANGED, {
-            'dateOrTimeFieldPicker': _me,
-            'newValue': newValue
-          });
-        }
-      },
-
-      _onChangeField: function(currentValue, data) {
-        const _me = this;
-        const value = currentValue ? $j.format.date(currentValue, _me._defaultFormat) : "";
-        let prototypejsEle = $(data[0]);
-        prototypejsEle.value = value;
-        console.debug('_onChangeField: ', value);
-        _me._onChanged();
+    get htmlElem() {
+      return this.#inputField;
+    }
+
+    get value() {
+      return this.htmlElem.value || '';
+    }
+
+    set value(value) {
+      this.htmlElem.value = value || '';      
+    }
+
+    setPickerConfig(config) {
+      $j(this.htmlElem).datetimepicker('setOptions',
+          Object.assign({}, this.#pickerConfig, config));
+    }
+
+    openPicker(event) {
+      event?.stop();
+      this.#openPickerNow = true;
+      $j(this.htmlElem).trigger('open');
+    }
+
+    #onPickerShow() {
+      const showNow = this.#openPickerNow;
+      this.#openPickerNow = false;
+      console.debug('#onPickerShow: ', showNow);
+      return showNow;
+    }
+
+    #onChanged() {
+      console.debug("#onChanged", this.value);
+      if (this.validate()) {
+        this.celFire(EVENT_FIELD_CHANGED, {
+          'dateOrTimeFieldPicker': this,
+          'newValue': this.value
+        });
+      }
+    }
+
+    #onPickerSelect(value) {
+      this.value = this.#marshaller.format(value);
+      this.#onChanged();
+    }
+
+    #getValidatedValue() {
+      if (!this.#fieldValidator) {
+        return this.value;
       }
+      const validated = this.#fieldValidator(this.value, {
+        min: this.#marshaller.parse(this.htmlElem.dataset.min),
+        max: this.#marshaller.parse(this.htmlElem.dataset.max)
+      });
+      return this.#marshaller.format(validated) || '';
+    }
+
+    validate() {
+      if (!this.value) {
+        return true;
+      }
+      const validatedValue = this.#getValidatedValue();
+      const valid = this.value === validatedValue;
+      this.htmlElem.classList.toggle('validation-failed', !valid);
+      if (!valid) {
+        console.info("validate: invalid", this.htmlElem, this.value, '->', validatedValue);
+        this.value = validatedValue;
+      }
+      return valid;
+    }
 
-    });
-    window.CELEMENTS.structEdit.DateOrTimeFieldPicker.prototype = Object.extend(
-      window.CELEMENTS.structEdit.DateOrTimeFieldPicker.prototype, CELEMENTS.mixins.Observable);
   }
 
   class CelementsDateTimePickerFactory {

```

Comment:
please change this class to "static". Overwrite the constructor with an exception throw that the class is static.
e.g. [constructor of static class](https://medium.com/@fab1o/javascript-static-class-3e1acdaec81)
Make all public methods static.
[mdn docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#static_methods_and_properties)